### ISSUE TITLE
Dropdown filters display as text fields once embedded

### ISSUE THREAD

**Issue by lonelyviking** [metabase/metabase#47286]
### Describe the bug

We're experiencing and issue where filters that are set as dropdowns in the Metabse UI are displaying as text fields when embedded in our app.

This is happening to multiple filters, but not to all of them. Additionally, one of the dropdown filters shows one option when there should be four.

Here are screenshots of the two scenarios:
![image](https://github.com/user-attachments/assets/3b1fea82-7190-4812-9493-7c5ef1bc68a0)
![image](https://github.com/user-attachments/assets/c8e6569b-62c6-4dc9-abb6-12fc8ecaf29a)

We have tried:

- Clearing the field caches in Admin > Table Metadata
- Adding custom values to the dropdown list
- Changing the table type to ENUM instead of Category
- Unlocking the Locked filter

Further details:

Database: Postgres
Embedding method: Python

Strangely it seems that the API actually returns an empty array for the filter values:
![image](https://github.com/user-attachments/assets/df446874-f971-4e8f-a408-f53e8a8938a4)

Entering a value that exists into the filter does work and results are filtered accordingly.

We also made a very basic dashboard with a basic query and only two filters to replicate this issue in the least complex scenario.

Here is the Metabase dash showing the two filters and the status filter correctly populated:
![image](https://github.com/user-attachments/assets/399bd20d-7841-4803-872e-18f9eb52392c)

Here is the embedded view showing the status filter with only one of three options (in other cases this shows as a text field):
![image](https://github.com/user-attachments/assets/ca105c30-cfd6-4848-9035-eada90e2d2e6)

I think we have narrowed it down to a specific scenario though.

Our app is a 'multi-tenanted' app with multiple organisations and we pre-filter results per org by dynamically passing in the org param via the backend. We make the org filter a locked param in the embed settings (screenshot below).

![image](https://github.com/user-attachments/assets/f76a1a7b-c71e-473c-bb6d-d9d733cf1595)

Making the org filter Editable in the embed settings doesn't make any difference.

If we remove the org filter from the query then the embedded version gets all the dropdown options.

Here is how we are handling the param injection via the backend:
![image](https://github.com/user-attachments/assets/5ff23757-3207-4ae3-8d3f-66a86db17217)

### To Reproduce

1. Create a dashboard with drop down (list of values) filters
2. Embed dashboard and inject params via backend (Python)
3. Drop downs are not populated fully or display as text fields - value route returns empty array

![image](https://github.com/user-attachments/assets/2ea4cd49-fe33-4787-b519-7046a6b759b9)

### Expected behavior

The value route should return the array of values and the drop downs should work as expected. 

### Logs

{" values" : [], "has more values": false}

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-ZA",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.24+8",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.24",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.24+8",
    "os.name": "Linux",
    "os.version": "5.15.0-1053-aws",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "16.1 (Debian 16.1-1.pgdg120+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    },
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2024-08-13",
      "tag": "v0.50.20",
      "hash": "df82d58"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

It's a relatively big impact on the end user.

**Comment by paoliniluis** [comment]
@lonelyviking when you embed, all filters behave as if they were linked together, to avoid data leakage. If you embed locking a filter, then all the other filters (no matter if they're linked or not) will be linked to the value you pass

**Comment by lonelyviking** [comment]
@paoliniluis Thanks for the feedback. That makes total sense. In this case all the other filters are supposed to be linked to the locked Org filter and should always return data (as they do outside of the embedded scenario and when the filter is not locked).

**Comment by akshayhubilo** [comment]
Any update here?

**Comment by lonelyviking** [comment]
> Any update here?

@akshayhubilo No, nothing new. All of the values we were dealing with happened to be enums that will also be the same no matter the Org filter value so we ended up just adding them as custom values to the dropdown filter element which has worked for us. Not ideal, but the dash is working for the users at least.

**Comment by ranquild** [comment]
Looks like a BE only bug since the FE requests field values and the BE (incorrectly) returns an empty list. While the author of the issue doesn't mention it, I expect that there is a complex permission setup here.

**Comment by paoliniluis** [comment]
@lonelyviking so the data type of all fields in the database are enums, is that correct? I would love to try to provide a clear reproduction here so we can fix it

**Comment by heypoom** [comment]
I tried to reproduce this issue by static embedding a dashboard, setting one of the parameters to "Locked" and other parameters to "Editable" in the embedding settings, then pass the values from the backend (using the Node.js script as-is, which should be equivalent to the Python script present in this ticket from what I can see)

![Image](https://github.com/user-attachments/assets/e0769434-f53d-4216-b1f1-fa71b898ccd5)

I'm still getting a correctly-filtered result, and I couldn't get the parameters to become text.

![Image](https://github.com/user-attachments/assets/252b0752-f14b-4f2b-a748-a8cc4cf768f2)

As Alex mentioned, I also suspect that this is partially caused by sandboxing or permissions setup in the instance. It would be great if we can see the sandboxing or permissions setup, so we can try to replicate that.

**Comment by perivamsi** [comment]
Moving this to P2 while we gather more details around this

**Comment by perivamsi** [comment]
@amandaflotelime if you have steps to reproduce this issue, that will be very helpful

**Comment by amandaflotelime** [comment]
@perivamsi I don't really have steps to reproduce it. BU ti could show you exemples if we book a meeting

**Comment by perivamsi** [comment]
@amandaflotelime I'll have someone reach out to you to book a meeting

**Comment by paoliniluis** [comment]
@amandaflotelime Luis here from the Success team. I met with your team several times and we would need you to upgrade Metabase as we have a very strong belief that this issue is already fixed by now

**Comment by paoliniluis** [comment]
Closing due to no response

**Comment by amandaflotelime** [comment]
Hello again, i checked with our developers and they're asking in which version this is fixed? :)

**Comment by amandaflotelime** [comment]
@paoliniluis Did you see my comment above? :)
